gossip/src/crds/geyser_plugin_utils.rs

```diff
@@ -0,0 +1,26 @@
+use crate::{
+    crds::{Crds, Pubkey, VersionedCrdsValue},
+    crds_value::CrdsData,
+};
+
+impl Crds {
+    /// Notified when a node cluster info is updated by gossip network.
+    pub fn notify_clusterinfo_update(&self, crd_value: Option<&VersionedCrdsValue>) {
+        if let Some(clusterinfo_update_notifier) = &self.clusterinfo_update_notifier {
+            if let Some(value) = crd_value {
+                if let CrdsData::LegacyContactInfo(ref cluster_info) = value.value.data {
+                    let notifier = &clusterinfo_update_notifier.read().unwrap();
+                    notifier.notify_clusterinfo_update(cluster_info);
```

Comment:
oh jk I see you already set it up to update only on reception of LegacyContactInfo